// zcf/dist/index.mjs
export { A as AI_OUTPUT_LANGUAGES, C as CLAUDE_DIR, e as CLAUDE_MD_FILE, f as ClAUDE_CONFIG_FILE, j as LANG_LABELS, L as LEGACY_ZCF_CONFIG_FILE, S as SETTINGS_FILE, h as SUPPORTED_LANGS, Z as ZCF_CONFIG_FILE, I as addCompletedOnboarding, w as applyAiLanguageDirective, n as backupExistingConfig, E as backupMcpConfig, G as buildMcpServerConfig, c as cleanupPermissions, a as commandExists, q as configureApi, p as copyConfigFiles, l as ensureClaudeDir, H as fixWindowsMcpConfig, k as getAiOutputLanguageLabel, v as getExistingApiConfig, t as getExistingModelConfig, z as getMcpConfigPath, g as getPlatform, b as importRecommendedEnv, d as importRecommendedPermissions, i as init, y as installClaudeCode, x as isClaudeCodeInstalled, m as mergeAndCleanPermissions, r as mergeConfigs, F as mergeMcpServers, s as mergeSettingsFile, o as openSettingsJson, B as readMcpConfig, u as updateDefaultModel, D as writeMcpConfig } from './chunks/simple-config.mjs';
import 'node:fs';
import 'node:process';
import 'ansis';
import 'inquirer';
import 'node:url';
import 'i18next';
import 'i18next-fs-backend';
import 'pathe';
import 'node:child_process';
import 'node:os';
import 'node:util';
import 'dayjs';
import 'ora';
import 'semver';
import 'tinyexec';
import 'node:fs/promises';
